app/react/Library/components/SortButtons.tsx

```diff
@@ -0,0 +1,199 @@
+  stateProperty: string;
+  // eslint-disable-next-line react/no-unused-prop-types
+  selectedTemplates: IImmutable<string[]>;
+  sortCallback: Function;
+}
+
+const mapStateToProps = (state: IStore, ownProps: SortButtonsOwnProps) => {
+  let templates;
+  const stateProperty = ownProps.stateProperty ? ownProps.stateProperty : 'library.search';
+
+  if (ownProps.selectedTemplates && ownProps.selectedTemplates.count()) {
+    templates = state.templates.filter(
+      i => i !== undefined && ownProps.selectedTemplates.includes(i.get('_id'))
+    )! as IImmutable<ClientTemplateSchema[]>;
+  }
+
+  const search = stateProperty
+    .split(/[.,/]/)
+    .reduce(
+      (memo: { [k: string]: any }, property: string) =>
+        Object.keys(memo).indexOf(property) !== -1 ? memo[property] : null,
+      state
+    );
+  return { stateProperty, search, templates: templates || state.templates };
+};
+
+const mapDispatchToProps = (dispatch: Dispatch<{}>) =>
+  bindActionCreators({ merge: actions.merge }, wrapDispatch(dispatch, 'library'));
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type mappedProps = ConnectedProps<typeof connector> & SortButtonsOwnProps;
+
+const getPropertySortType = (selected: SortType): string =>
+  selected.type === 'text' || selected.type === 'select' ? 'string' : 'number';
+const getToggleSearchIcon = (search: SearchOptions) =>
+  search.order === 'asc' && search.sort !== '_score' ? 'arrow-up' : 'arrow-down';
+
+const defaultSorts = [
+  { label: 'Title', value: 'title', type: 'text', context: 'System' },
+  { label: 'Date added', value: 'creationDate', type: 'number', context: 'System' },
+  { label: 'Date modified', value: 'editDate', type: 'number', context: 'System' },
+];
+
+const getCommonSorts = (search: SearchOptions) => [
+  ...defaultSorts,
+  ...(search.searchTerm
+    ? [
+        {
+          label: 'Search relevance',
+          value: '_score',
+          type: 'number',
+          context: 'System',
+        },
+      ]
+    : []),
+];
+
+const validateSearch = (search: SearchOptions): SearchOptions =>
+  search.sort === '_score' && !search.searchTerm
+    ? {
+        sort: 'creationDate',
+        order: 'desc',
+        searchTerm: search.searchTerm,
+      }
+    : { searchTerm: search.searchTerm, sort: search.sort };
+
+const sortDirection = (condition: string) => {
+  switch (condition) {
+    case 'desc':
+    case 'string':
+      return 'asc';
+    case 'asc':
+    default:
+      return 'desc';
+  }
+};
+
+const SortButtonsComponent = ({
+  search,
+  merge,
+  stateProperty,
+  templates,
+  sortCallback,
+}: mappedProps) => {
+  const doSort = (property: string, defaultTreatAs: string, selectedSort?: string) => {
+    const treatAs = defaultTreatAs;
+    const order = selectedSort || sortDirection(treatAs);
+    const newSort = { sort: property, order, treatAs };
+    merge(stateProperty, newSort);
+    const filters = { ...search, ...newSort, userSelectedSorting: true };
```

Comment:
Yo can copy newSort with `...omit(newSort, 'treatAs')`and avoid the delete later, this returns the correct type for filters too.

Reply:
done